File: ws-client/src/main/java/net/devopssolutions/demo/ws/client/component/WsProducer.java
package net.devopssolutions.demo.ws.client.component;

import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.extern.slf4j.Slf4j;
import net.devopssolutions.demo.ws.rpc.RpcMessage;
import net.devopssolutions.demo.ws.rpc.RpcMethods;
import net.devopssolutions.demo.ws.rpc.RpcType;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import javax.websocket.Session;
import java.io.IOException;
import java.io.OutputStream;
import java.util.UUID;
import java.util.zip.GZIPOutputStream;

@Slf4j
@Component
public class WsProducer {

    private ObjectMapper objectMapper = new ObjectMapper();

    @Autowired
    private WsConsumer wsConsumer;

    @Scheduled(fixedRate = 10_000)
    private void sendHello() {
        Session session = wsConsumer.getSession().get();
        log.info("sending hello: {}", session != null);
        if (session != null) {
            try {
                RpcMessage<Object, Object> message = RpcMessage.builder()
                        .id(UUID.randomUUID().toString())
                        .method(RpcMethods.HELLO)
                        .type(RpcType.REQUEST)
                        .params("word")
                        .build();
                sendMessage(session, message);
            } catch (Exception e) {
                log.warn("exception sending hello", e);
            }
        }
    }

    @Scheduled(fixedRate = 30_000, initialDelay = 10_000)
    private void sendFlood() {
        Session session = wsConsumer.getSession().get();
        log.info("sending hello: {}", session != null);
        if (session != null) {
            try {
                RpcMessage<Object, Object> message = RpcMessage.builder()
                        .id(UUID.randomUUID().toString())
                        .method(RpcMethods.FLOOD)
                        .type(RpcType.REQUEST)
                        .params(1_000_000)
                        .build();
                sendMessage(session, message);
            } catch (Exception e) {
                log.warn("exception sending hello", e);
            }
        }
    }

    private void sendMessage(Session session, RpcMessage<Object, Object> message) throws IOException {
        OutputStream sendStream = null;
        try {
            sendStream = session.getBasicRemote().getSendStream();
            objectMapper.writeValue(new GZIPOutputStream(sendStream), message);
        } catch (IOException e) {
            if (sendStream != null) {
                sendStream.close();
            }
            throw e;
        }
    }

}

File: ws-client/src/main/java/net/devopssolutions/demo/ws/client/component/WsProducer.kt
package net.devopssolutions.demo.ws.client.component

import com.fasterxml.jackson.databind.ObjectMapper
import mu.KLogging
import net.devopssolutions.demo.ws.rpc.RpcMessage
import net.devopssolutions.demo.ws.rpc.RpcMethods
import net.devopssolutions.demo.ws.rpc.RpcType
import org.springframework.stereotype.Component
import org.springframework.web.reactive.socket.WebSocketMessage
import org.springframework.web.reactive.socket.WebSocketSession
import reactor.core.publisher.Flux
import java.time.Duration
import java.time.LocalDateTime
import java.time.ZoneOffset
import java.util.*
import java.util.zip.GZIPOutputStream

@Component
open class WsProducer(
        private val objectMapper: ObjectMapper
) {
    companion object : KLogging()


    fun sendHello(session: WebSocketSession): Flux<WebSocketMessage> {
        return Flux.interval(Duration.ofSeconds(0), Duration.ofSeconds(30))
                .map { helloMessage(it) }
                .map { message -> buildMessage(session, message) }
    }

    private fun helloMessage(number: Long): RpcMessage<String, Unit> = RpcMessage(
            id = UUID.randomUUID().toString(),
            created = LocalDateTime.now(ZoneOffset.UTC),
            method = RpcMethods.HELLO.method,
            type = RpcType.REQUEST,
            params = "word $number")

    fun sendFlood(session: WebSocketSession): Flux<WebSocketMessage> {
        return Flux.interval(Duration.ofSeconds(10), Duration.ofSeconds(30))
                .map { floodMessage(it) }
                .map { message -> buildMessage(session, message) }
    }

    private fun floodMessage(number: Long): RpcMessage<Int, Unit> = RpcMessage(
            id = UUID.randomUUID().toString(),
            created = LocalDateTime.now(ZoneOffset.UTC),
            method = RpcMethods.FLOOD.method,
            type = RpcType.REQUEST,
            params = 1_000_000)

    private fun buildMessage(session: WebSocketSession, message: RpcMessage<*, Unit>): WebSocketMessage =
            session.binaryMessage {
                val buffer = it.allocateBuffer()
                objectMapper.writeValue(GZIPOutputStream(buffer.asOutputStream()), message)
                buffer
            }

}
